### PR TITLE
Add SSL certificate check handling

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Depends on https://github.com/nim-lang/Nim/pull/6664

### CODE REVIEW

File: src/nimblepkg/publish.nim [hunk 1]
```diff
@@ -51,7 +52,12 @@ proc requestNewToken(cfg: Config): string =
 
 proc getGithubAuth(o: Options): Auth =
   let cfg = o.config
-  result.http = newHttpClient(proxy = getProxy(o))
+  var sslVerifyMode = CVerifyPeer
+  if o.disableSslCertCheck:
+    display("Warning:", "disabling SSL certificate checking", Warning)
+    sslVerifyMode = CVerifyNone
+  let ctx = newContext(verifyMode = sslVerifyMode)
```
Comment (dom96) [inline src/nimblepkg/publish.nim]: This code is repeated and can be refactored out into a function :)

**Comment by dom96** [comment]
Waiting on fixes to feedback and on  nim-lang/Nim#6664.

**Comment by genotrance** [comment]
Looks like the upstream changes have been merged in https://github.com/nim-lang/Nim/pull/13697.

@FedericoCeratto - do you think you will be able to continue work on this PR?

**Comment by FedericoCeratto** [comment]
@genotrance yes, I'm going to rebase and update it

**Comment by FedericoCeratto** [comment]
@dom96 the certificate check is passing on my hosts but failing on travis ci

File: src/nimblepkg/packageinfo.nim [hunk 2]
```diff
@@ -199,8 +200,15 @@ proc fetchList*(list: PackageList, options: Options) =
...
       try:
-        let client = newHttpClient(proxy = proxy)
+        let ctx = newSSLContext(options.disableSslCertCheck)
+        let client = newHttpClient(proxy = proxy, sslContext = ctx)
         client.downloadFile(url, tempPath)
+      except SslError:
+        let message = "Failed to verify the SSL certificate for " & url
+        display("Error:", message, Error, HighPriority)
+        # Terminate immediately: don't encourage ignoring the error
+        quit(QuitFailure)
```
Comment (dom96) [inline src/nimblepkg/packageinfo.nim:210]: CC @genotrance, be aware that you should never quit in Nimble code but instead use this API.

```suggestion
        let message = "Failed to verify the SSL certificate for " & url
        raiseNimbleError(message, "Use --noSSLCheck to ignore this error.")
```

Comment (FedericoCeratto) [reply src/nimblepkg/packageinfo.nim]: Updated

**Comment by dom96** [comment]
@FedericoCeratto can you rebase? Maybe that will get this passing CI :)

**Comment by FedericoCeratto** [comment]
@dom96 rebased

**Comment by dom96** [comment]
Thank you!

**Comment by FedericoCeratto** [comment]
Tested by setting `url = "https://expired.badssl.com/"` in `~/.config/nimble/nimble.ini`, or `https://wrong.host.badssl.com/` or `https://untrusted-root.badssl.com/`. It prevents the download and shows `Hint: Use --noSSLCheck to ignore this error.`

**Pull request merged**